test/functional/custom-transform.spec.ts

```diff
@@ -66,62 +125,51 @@ describe("custom transformation decorator", () => {
 
     });
 
-    it("versions and groups should work with @Transform decorator too", () => {
+    it("Non class objects should not break if transform is used", () => {
```

Comment:
what is the point to use `classToPlain` if your object is not instance of some class? Because thats what `classToPlain` supposed to do - convert class to plain.